rows_test.go

```diff
@@ -18,52 +18,59 @@ func assert(test_val interface{}, expected_val interface{}, t *testing.T, err st
-		"	{\"name\":\"timestampntz_col\",\"type\":\"TIMESTAMPNTZ\"}," +
-		"	{\"name\":\"timestamptz_col\",\"type\":\"TIMESTAMPTZ\"}," +
-		"	{\"name\":\"legacy_bool_col\",\"type\":\"UInt8\"}," +
-		"	{\"name\":\"array_col\",\"type\":\"Array(Int32)\"}," +
-		"	{\"name\":\"nested_array_col\",\"type\":\"Array(Array(String))\"}," +
-		"	{\"name\":\"new_bool_col\",\"type\":\"Boolean\"}]," +
-		"\"data\":[" +
-		"	[null,1,0.312321,123213.321321,\"text\", \"2080-12-31\",\"1989-04-15 01:02:03\",\"0002-01-01\",\"1989-04-15 01:02:03.123456\",\"1989-04-15 02:02:03.123456+00\",1,[1,2,3],[[]],true]," +
-		"	[2,1,0.312321,123213.321321,\"text\",\"1970-01-01\",\"1970-01-01 00:00:00\",\"0001-01-01\",\"1989-04-15 01:02:03.123457\",\"1989-04-15 01:02:03.1234+05:30\",1,[1,2,3],[[]],true]," +
-		"	[3,null,0.312321,123213.321321,\"text\",\"1970-01-01\",\"1970-01-01 00:00:00\",\"0001-01-01\",\"1989-04-15 01:02:03.123458\",\"1989-04-15 01:02:03+01\",1,[5,2,3,2],[[\"TEST\",\"TEST1\"],[\"TEST3\"]],false]," +
-		"	[2,1,0.312321,123213.321321,\"text\",\"1970-01-01\",\"1970-01-01 00:00:00\",\"0001-01-01\",\"1989-04-15 01:02:03.123457\",\"1111-01-05 17:04:42.123456+05:53:28\",1,[1,2,3],[[]],false]," +
-		"	[2,1,0.312321,123213.321321,\"text\",\"1970-01-01\",\"1970-01-01 00:00:00\",\"0001-01-01\",\"1989-04-15 01:02:03.123457\",\"1989-04-15 02:02:03.123456-01\",1,[1,2,3],[[]],null]" +
-		"]," +
-		"\"rows\":5," +
-		"\"statistics\":{" +
-		"	\"elapsed\":0.001797702," +
-		"	\"rows_read\":3," +
-		"	\"bytes_read\":293," +
-		"	\"time_before_execution\":0.001251613," +
-		"	\"time_to_execute\":0.000544098," +
-		"	\"scanned_bytes_cache\":2003," +
-		"	\"scanned_bytes_storage\":0}}", "" +
-		"{" +
-		"\"query\":{\"query_id\":\"16FF2A0300ECA753\"}," +
-		"\"meta\":[{\"name\":\"int_col\",\"type\":\"Nullable(Int32)\"}]," +
-		"\"data\":[[3], [null]]," +
-		"\"rows\":2," +
-		"\"statistics\":{" +
-		"	\"elapsed\":0.001797702," +
-		"	\"rows_read\":2," +
-		"	\"bytes_read\":293," +
-		"	\"time_before_execution\":0.001251613," +
-		"	\"time_to_execute\":0.000544098," +
-		"	\"scanned_bytes_cache\":2003," +
-		"	\"scanned_bytes_storage\":0}}"}
+	resultJson := []string{
+		`{
+        "query":{"query_id":"16FF2A0300ECA753"},
+        "meta":[
+        	{"name":"int_col","type":"int null"},
+        	{"name":"bigint_col","type":"long"},
+        	{"name":"float_col","type":"float"},
+        	{"name":"double_col","type":"double"},
+        	{"name":"text_col","type":"text"},
+        	{"name":"date_col","type":"date"},
+        	{"name":"timestamp_col","type":"timestamp"},
+        	{"name":"pgdate_col","type":"pgdate"},
+        	{"name":"timestampntz_col","type":"timestampntz"},
+        	{"name":"timestamptz_col","type":"timestamptz"},
+        	{"name":"legacy_bool_col","type":"int"},
+        	{"name":"array_col","type":"array(int)"},
+        	{"name":"nested_array_col","type":"array(array(text))"},
+        	{"name":"new_bool_col","type":"boolean"}],
```

Comment:
I think it might be worth testing Decimal with scale/precision and array of decimal (like: array(Decimal(38, 30)) )